### PR TITLE
Generate BubbleProf and Flamegraph on CI

### PR DESCRIPTION
**Pull request by lalexgap**
Generates a flamegraph and bubbleprof on CI to help us investigate performance.

Example: https://app.circleci.com/pipelines/github/statechannels/statechannels/8769/workflows/a3062384-6ce8-4a7c-aff5-5e11a4f9a09a/jobs/41409/artifacts

Closes https://github.com/statechannels/statechannels/issues/2407

### CODE REVIEW

**Comment by lalexgap** [comment]
It was showing me a failing `prepare` step but it looks ok here: https://app.circleci.com/pipelines/github/statechannels/statechannels?branch=profiling

It seems to be pointing to a canceled build.

File: .circleci/config.yml [hunk 1]
```diff
@@ -294,6 +294,12 @@ jobs:
           command: yarn test:stress
           no_output_timeout: 10m
       - upload_artifacts
+      - store_artifacts:
+          path: /home/circleci/project/packages/server-wallet/profiling-data/stress-test-1-payment.html
+          destination: stress-test-1-payment.html
+      - store_artifacts:
+          path: /home/circleci/project/packages/server-wallet/profiling-data/stress-test-25-payment.html
+          destination: stress-test-25-payment.html
```
Comment (andrewgordstewart) [inline .circleci/config.yml:302]:
```suggestion
      - store_artifacts:
          path: /home/circleci/project/packages/server-wallet/profiling-data
```

We used to have a mess of uploading a dozen different artifacts. It's much cleaner to just upload an entire directory.

(Even better might be to just put the artifacts in the `artifacts` folder that is currently uploaded?)

File: packages/server-wallet/e2e-test/e2e-utils.ts [hunk 2]
```diff
@@ -1,34 +1,53 @@
...
-    console.log(PerformanceTimer.formatResults(JSON.parse(message as any)))
+  const bubbleprof = new ClinicBubbleprof({dest: PROFILE_DATA_PATH});
+  return new Promise((resolve: any): void =>
+    bubbleprof.collect(
+      ['ts-node', join(__dirname, '/payer/index.ts'), ...args],
+      (err: any, fileName: string) => {
+        if (err) throw err;
+        resolve(fileName);
+      }
+    )
```
Comment (andrewgordstewart) [inline packages/server-wallet/e2e-test/e2e-utils.ts:49]: Do you know how much, if any, overhead `bubbleprof.collect` causes?

File: packages/server-wallet/e2e-test/stress.test.ts [hunk 3]
```diff
@@ -110,15 +118,17 @@ describe('Stress tests', () => {
       alice().privateKey,
       getParticipant('receiver', bob().privateKey),
       bob().privateKey,
-      100
+      2
```
Comment (andrewgordstewart) [inline packages/server-wallet/e2e-test/stress.test.ts:121]: Is this change intentional?

Comment (lalexgap) [reply packages/server-wallet/e2e-test/stress.test.ts:121]: Ugh nope!

Comment (lalexgap) [inline packages/server-wallet/e2e-test/stress.test.ts:121]:
```suggestion
      100
```

File: .circleci/config.yml (hunk 1 above)
Comment (lalexgap) [reply .circleci/config.yml:302]: There is also captured raw data in `profiling-data` that we don't want to upload but just storing the html files in the artifact folder makes sense.

File: packages/server-wallet/e2e-test/stress.test.ts [hunk 4]
```diff
@@ -93,15 +99,17 @@ describe('Stress tests', () => {
       alice().privateKey,
       getParticipant('receiver', bob().privateKey),
       bob().privateKey,
-      100
+      10
```
Comment (lalexgap) [inline packages/server-wallet/e2e-test/stress.test.ts:102]:
```suggestion
      100
```
